lib/spack/spack/ci.py

```diff
@@ -1268,18 +1249,19 @@ def main_script_replacements(cmd):
                     build_stamp = cdash_handler.build_stamp
                     job_vars["SPACK_CDASH_BUILD_STAMP"] = build_stamp
 
-                variables.update(job_vars)
-
-                artifact_paths = [
-                    rel_job_log_dir,
-                    rel_job_repro_dir,
-                    rel_job_test_dir,
-                    rel_user_artifacts_dir,
-                ]
+                job_object["artifacts"] = {
```

Comment:
I can think of a use case for `needs`, but for artifacts I can imagine a case where users create custom artifact directories for jobs that would need this. Should this utilize the `spack.config.merge_yaml` instead?

Reply:
The original overwrote `artifacts` (and so I kept that logic), but I do see reason in allowing customization here. Can't think of a case where I personally would use it, but I can add that.